docs(forms): add migration guide for Signal Forms

kirjs opened the pull request:
A new guide for migrating from legacy forms (Reactive and Template-driven) to the new Signal Forms.

kirjs committed:
docs(forms): add migration guide for Signal Forms

This commit introduces a new guide for migrating from legacy forms (Reactive and Template-driven) to the new Signal Forms.

JeanMeche commented on adev/src/content/guide/forms/signals/migration.md:
````diff
@@ -0,0 +1,192 @@
+```typescript
+import {signal, Injector, inject} from '@angular/core';
+import {FormControl, Validators} from '@angular/forms';
+import {compatForm} from '@angular/forms/signals';
````
```suggestion
import {compatForm} from '@angular/forms/signals/compat';
```

JeanMeche commented on adev/src/content/guide/forms/signals/migration.md:
```diff
@@ -0,0 +1,192 @@
+
+* Complex asynchronous logic.
+* Intricate RxJS operators that are not yet ported.
+* Integration with legacy third-party libraries.
```
we might want to introduce `compatForm` a bit.

kirjs committed:
fixup! docs(forms): add migration guide for Signal Forms

kirjs committed:
docs(forms): add a Accessing values section

This explains how to get the value of a compat form

kirjs committed:
fixup! docs(forms): add a Accessing values section

kirjs committed:
docs(forms): Add a section for automatic form binding

To avoid manually adding classes like `.ng-valid`, `.ng-dirty` etc to every field, you can provide a global configuration using `provideSignalFormsConfig`.

leonsenft commented on adev/src/content/guide/forms/signals/migration.md:
```diff
@@ -0,0 +1,235 @@
+      @for (error of f.password().errors(); track error) {
+      <p>{{ error.message || error.kind }}</p>
+      }
+    </div>
```
Same here:
```suggestion
      <div class="error-list">
        @for (error of f.password().errors(); track error) {
          <p>{{ error.message || error.kind }}</p>
        }
      </div>
```

leonsenft commented on adev/src/content/guide/forms/signals/migration.md:
```diff
@@ -0,0 +1,235 @@
+      <input [field]="f.email">
+    </label>
+
+    @if (f.email().touched() && f.email().invalid()) {
```
Since the example above doesn't actually have a validation rule applied to the `email` field, can we just remove this to simplify the code snippet?

leonsenft commented on adev/src/content/guide/forms/signals/migration.md:
```diff
@@ -0,0 +1,235 @@
+### Integrating a `FormGroup` into a signal form
+
+You can also wrap an entire `FormGroup`. This is common when a reusable sub-section of a form—such as an **Address Block
+**—is still managed by legacy Reactive Forms.
```
Formatting on the preview site is broken here. I suspect due to the space before the final `**`:
```suggestion
You can also wrap an entire `FormGroup`. This is common when a reusable sub-section of a form—such
as an **Address Block**—is still managed by legacy Reactive Forms.
```

leonsenft commented on adev/src/content/guide/forms/signals/migration.md:
```diff
@@ -0,0 +1,235 @@
+      <div class="error-list">
+        <p>Customer name is required.</p>
+      </div>
+    }
```
Same suggestion to remove this here, since it's irrelevant to the example and there's no validation applied to the field anyways.

kirjs committed:
docs(forms): Address feedback

Change wording and formatting and clean up.

kirjs committed:
docs(forms): Address feedback

Update the Automatic status classes section to make NG_STATUS_CLASSES go first.

leonsenft commented on adev/src/content/guide/forms/signals/migration.md:
````diff
@@ -0,0 +1,247 @@
+behavior:
+
+```typescript
+import {provideSignalFormsConfig} from '@angular/forms/signals';
````
```suggestion
import {NG_STATUS_CLASSES, provideSignalFormsConfig} from '@angular/forms/signals';
```

leonsenft commented on adev/src/content/guide/forms/signals/migration.md:
```diff
@@ -0,0 +1,247 @@
+
+Reactive forms used to bind classes like `.ng-valid` or `.ng-dirty` to every field. Signal forms do not do that anymore.
+
+If you want this behavior you can provide an NG_STATUS_CLASSES preset that can be provided to match the Reactive Forms
```
nit: phrasing (lots of duplication between "want this behavior"/"match the [behavior]" and "you can provide"/"can be provided".

```suggestion
If you want to preserve this behavior, you can provide the `NG_STATUS_CLASSES` preset:
```

kirjs committed:
docs(forms): Address feedback

Update `Automatic status class` section

bencodezen reviewed (approved):
Thanks for creating this guide @kirjs! Excited to get it into users' hands.

kirjs committed:
docs(forms): update migration guide

ShacharHarshuv commented:
Hey, I couldn't find the mentioned `NG_STATUS_CLASSES` object  and I believe I'm on the version, what did I miss?
